primitives/src/merkle_tree/field_based_mht/batch_mht/mod.rs

```diff
@@ -0,0 +1,30 @@
+use crate::{BatchFieldBasedHash};
+
+pub mod poseidon;
+
+pub trait BatchMerkleTree:Clone {
+    type Hash: BatchFieldBasedHash;
+
+    // This function reset the pointing indices for the new element positions and processed positions
+    // Mainly used for benchmarking
+    fn reset(&mut self);
+
+    // This function adds a leaf to the Merkle tree
+    // Whenever the number of added leafs reaches the block size passed as parameter during creation of the
+    // Merkle tree, the parent nodes are updated with the corresponding hashes
+    fn update(&mut self, leaf: <Self::Hash as BatchFieldBasedHash>::Data);
```

Comment:
Also, it's true that the leaf, inside, will be copied anyway, but probably it's worth to pass a reference here and not the entire Field instance: the copy of a reference it's much cheaper than the copy of a Field